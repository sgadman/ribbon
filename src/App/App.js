import React, { useEffect } from "react";
import { Router, Route, Switch } from "react-router-dom";

import { makeStyles } from "@material-ui/styles";

import Ribbon from "../components/Ribbon";
import history from "../components/Utils/history";
import Home from "../pages/Home";
import Bug from "../pages/Bug";
import Epic from "../pages/Epic";

const useStyles = makeStyles(() => ({
  app: {
    textAlign: "center",
    height: "100vh",
    width: "100vw",
    display: "flex",
    flexDirection: "row"
  }
}));

const App = () => {
  const classes = useStyles({});

  useEffect(() => {
    history.push("home");
  });

  return (
    <div className={classes.app}>
      <Ribbon />
      <Router basename="/pega/" history={history}>
        <div style={{marginLeft: 70}}>
          <Switch>
            <Route path="/home" component={Home} />
            <Route path="/bug" component={Bug} />
            <Route path="/epic" component={Epic} />
          </Switch>
        </div>
      </Router>
    </div>
  );
};

export default App;
