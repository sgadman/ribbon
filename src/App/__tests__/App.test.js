import React from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

import App from "../App";

describe("<App />", () => {
  let wrapper = null;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it("should match snapshot", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
