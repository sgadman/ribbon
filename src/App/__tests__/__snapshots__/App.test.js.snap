// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<div
  className="makeStyles-app-1"
>
  <Ribbon />
  <Router
    basename="/pega/"
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <div
      style={
        Object {
          "marginLeft": 60,
        }
      }
    >
      <Switch>
        <Route
          component={[Function]}
          path="/home"
        />
        <Route
          component={[Function]}
          path="/bug"
        />
        <Route
          component={[Function]}
          path="/epic"
        />
      </Switch>
    </div>
  </Router>
</div>
`;
